packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -284,8 +284,8 @@ class TextSelectionOverlay {
   void showHandles() {
     assert(_handles == null);
     _handles = <OverlayEntry>[
-      OverlayEntry(builder: (BuildContext context) => _buildHandle(context, _TextSelectionHandlePosition.start)),
-      OverlayEntry(builder: (BuildContext context) => _buildHandle(context, _TextSelectionHandlePosition.end)),
+      OverlayEntry(builder: (_) => _buildHandle(context, _TextSelectionHandlePosition.start)),
+      OverlayEntry(builder: (_) => _buildHandle(context, _TextSelectionHandlePosition.end)),
```

Comment:
this won't work reliably. When you look up the Theme with Theme.of, it causes the relevant Theme to register the context so that when the Theme changes, the context is marked dirty. But that won't work here because that would mark the widget whose context was passed to TextSelectionOverlay dirty, but not the handles. So the handles wouldn't change color.

(When you use the build context of the OverlayEntry, as in the original code, then when the relevant Theme changes, it causes the overlay entry to rebuild, rebuilding the handle.)